CONTRIBUTING.md

```diff
@@ -57,11 +57,30 @@ Addresses #bugnumber (in this specific format)
 Your pull request should:
 
 * Include a description of what your change intends to do
+  * The title of your PR should be a very brief description of the change, and
+    can use tags to allow for speedy categorization.
+    * Title tags are generally one word or acronym enclosed in square brackets.
+      Limiting to one or two tags is ideal to keep titles short. Some examples
+      of common tags are:
+      * [NFC] - No Functional Change
+      * [RFC] - Request For Comments (often used for drafts to get feedback)
+      * [Doc] - Documentation change
+      * [SPIRV] - Changes related to SPIR-V
+      * [HLSL2021] - Changes related to HLSL 2021 features
+      * Other tags in use: [Linux], [mac], [Win], [PIX], etc...
+      * Tags aren't formalized or any specific limited set. If you're unsure of
+        a reasonable tag to use, just don't use any. If you want to invent a new
+        tag, go for it! These are to help categorize changes at a glance.
+  * The PR description should include a more detailed description of the change,
+    an explanation for the motivation of the change, and links to any relevant
+    Issues. This does not need to be a dissertation, but should leave
+    breadcrumbs for the next person debugging your code (who might be you).
```

Comment:
Also mentioned in the template above, but I can't comment that far outside of a change. 

I like how if you inclide `fixes #1234` then it will close that issue. I don't think `addresses #1234` does anything except link to the issue, which is good. Do we want to encourage that in cases where it is relevant?